application/app/services/nav/main_item.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module Nav
+  class MainItem
+    attr_reader :id, :label, :url, :items, :position, :hidden, :type, :alignment, :new_tab, :icon, :partial
+
+    def initialize(id:, label: nil, url: nil, items: nil, position: nil, hidden: nil, alignment: nil, new_tab: nil, icon: nil, partial: nil)
+      @id = id
+      @label = label
+      @url = url
+      @items = items&.map { |child| Nav::MenuItem.new(**child) }
+      @position = position
+      @hidden = hidden.nil? ? false : hidden
+      @type = infer_type
+      @alignment = alignment || 'left'
+      @new_tab = new_tab.nil? ? false : new_tab
+      @icon = icon
+      @partial = partial
```

Comment:
we should validate that the partial path is part of the Rails project and not a malicious path accessing unauthorized files, because this value is injected directly into the view to be rendered